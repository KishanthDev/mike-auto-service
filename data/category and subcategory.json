[
  {
    "category": "Automotive",
    "subcategories": ["Car Repair", "Car Sales", "Tires", "Auto Parts"]
  },
  {
    "category": "Health & Medical",
    "subcategories": [
      "Hospitals",
      "Clinics",
      "Dentists",
      "Pharmacies",
      "Physiotherapy Centres",
      "Veterinary Clinics"
    ]
  },
  {
    "category": "Education",
    "subcategories": [
      "Schools (primary, secondary)",
      "Colleges & Universities",
      "Tutoring Services",
      "Educational Consultants"
    ]
  },
  {
    "category": "Restaurants & Food",
    "subcategories": [
      "Fast Food",
      "Fine Dining",
      "Cafes",
      "Bakeries",
      "Food Delivery Services"
    ]
  },
  {
    "category": "Shopping & Retail",
    "subcategories": [
      "Clothing Stores",
      "Electronics Stores",
      "Jewellery Showrooms",
      "Grocery Stores",
      "Department Stores",
      "Furniture Showrooms",
      "Pest Control"
    ]
  },
  {
    "category": "Professional Business Services",
    "subcategories": [
      "Chartered Accountants",
      "Lawyers",
      "Marketing Agencies",
      "Business Consultants"
    ]
  },
  {
    "category": "Home Services",
    "subcategories": [
      "Interior Designers",
      "Carpenters",
      "Plumbers",
      "Electricians",
      "Cleaners",
      "Landscapers",
      "Home Renovation"
    ]
  },
  {
    "category": "Travel & Transportation",
    "subcategories": [
      "Airlines",
      "Hotels",
      "Car Rentals",
      "Travel Agencies",
      "Public Transport"
    ]
  },
  {
    "category": "Real Estate",
    "subcategories": [
      "Builders and Developers",
      "Civil Contractors",
      "Real Estate Agents",
      "Property Management"
    ]
  },
  {
    "category": "Financial Services",
    "subcategories": [
      "Banks",
      "Loan Agencies",
      "Insurance Agencies",
      "Investment Services",
      "Credit Unions"
    ]
  },
  {
    "category": "Government & Community",
    "subcategories": [
      "Local Government Offices",
      "Community Centers",
      "Libraries",
      "Post Offices"
    ]
  },
  {
    "category": "Entertainment & Media",
    "subcategories": [
      "Movie Theaters",
      "Music Stores",
      "Event Management",
      "Convention Halls",
      "Media Production"
    ]
  },
  {
    "category": "Sports & Recreation",
    "subcategories": [
      "Gyms",
      "Sports Leagues",
      "Parks",
      "Recreation Centers",
      "Sports Equipment Stores"
    ]
  },
  {
    "category": "Pets & Animals",
    "subcategories": [
      "Pet Stores",
      "Veterinarians",
      "Pet Grooming",
      "Animal Shelters",
      "Pet Training"
    ]
  },
  {
    "category": "Computers & Mobile",
    "subcategories": [
      "Internet Service Providers",
      "Computer Sales and Repair",
      "Mobile Sales and Repair",
      "Web Design",
      "Software Solutions"
    ]
  },
  {
    "category": "Safety & Security",
    "subcategories": [
      "Private Security Services",
      "Fire Departments",
      "Police Stations",
      "Safety Equipment"
    ]
  }
]
