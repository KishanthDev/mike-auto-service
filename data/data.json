{
    "businessHours": [
      { "day": "Monday - Friday", "hours": "7:30 AM - 6:00 PM" },
      { "day": "Saturday", "hours": "8:00 AM - 4:00 PM" },
      { "day": "Sunday", "hours": "Closed" }
    ],
    "certifications": ["AUDI", "BMW", "Honda", "Toyota", "Ford"],
    "images": [
      "https://via.placeholder.com/150/92c952",
      "https://via.placeholder.com/150/92c952",
      "https://via.placeholder.com/150/92c952",
      "https://via.placeholder.com/150/92c952"
    ],
    "reviews": [
      {
        "author": "Sarah J.",
        "rating": 5,
        "content": "Mike's team fixed my Honda's transmission issue quickly and for much less than the dealership quoted. Very honest and transparent about costs."
      },
      {
        "author": "David T.",
        "rating": 5,
        "content": "I've been coming here for 5 years and have always had excellent service. They explain everything clearly and never push unnecessary repairs."
      }
    ],
    "services": [
      {
        "title": "Brake Service",
        "description": "Complete brake system inspection, repair and replacement. Lifetime warranty on most brake pads.",
        "price": "Starting at $89.99"
      },
      {
        "title": "Engine Repair",
        "description": "Diagnostic and repair services for all engine types. Factory-trained technicians.",
        "price": "Diagnosis: $89.99"
      },
      {
        "title": "Transmission Service",
        "description": "Transmission fluid change, repair and rebuilding services with 24-month warranty.",
        "price": "Fluid change: $149.99"
      },
      {
        "title": "Electrical Systems",
        "description": "Computer diagnostics, battery service, alternator replacement and more.",
        "price": "Starting at $69.99"
      }
    ]
  }
  