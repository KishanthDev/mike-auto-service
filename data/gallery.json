[
    "https://via.placeholder.com/150/92c952",
    "https://via.placeholder.com/150/92c952",
    "https://via.placeholder.com/150/92c952",
    "https://via.placeholder.com/150/92c952"
  ]
  