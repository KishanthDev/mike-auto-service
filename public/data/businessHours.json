[
    { "day": "Monday - Friday", "hours": "7:30 AM - 6:00 PM" },
    { "day": "Saturday", "hours": "8:00 AM - 4:00 PM" },
    { "day": "Sunday", "hours": "Closed" }
  ]
  