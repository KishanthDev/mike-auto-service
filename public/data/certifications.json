["AUDI", "BMW", "Honda", "Toyota", "Ford"]
