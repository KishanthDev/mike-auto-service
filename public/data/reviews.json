[
    {
      "author": "Sarah J.",
      "rating": 5,
      "content": "Mike's team fixed my Honda's transmission issue quickly and for much less than the dealership quoted. Very honest and transparent about costs."
    },
    {
      "author": "David T.",
      "rating": 5,
      "content": "I've been coming here for 5 years and have always had excellent service. They explain everything clearly and never push unnecessary repairs."
    }
  ]
  