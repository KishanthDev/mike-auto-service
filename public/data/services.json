[
    {
      "title": "Brake Service",
      "description": "Complete brake system inspection, repair and replacement. Lifetime warranty on most brake pads.",
      "price": "Starting at $89.99"
    },
    {
      "title": "Engine Repair",
      "description": "Diagnostic and repair services for all engine types. Factory-trained technicians.",
      "price": "Diagnosis: $89.99"
    },
    {
      "title": "Transmission Service",
      "description": "Transmission fluid change, repair and rebuilding services with 24-month warranty.",
      "price": "Fluid change: $149.99"
    },
    {
      "title": "Electrical Systems",
      "description": "Computer diagnostics, battery service, alternator replacement and more.",
      "price": "Starting at $69.99"
    }
  ]
  