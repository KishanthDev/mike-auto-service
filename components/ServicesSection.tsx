import { Card, CardHeader } from "@heroui/card";
import services from "../data/services.json";

export default function ServicesSection() {
  return (
    <section>
      <h2 className="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#2765c3]">
        Our Services
      </h2>
      <div className="grid grid-cols-1 sm:grid-cols-2 gap-4 mt-4 md:mt-6">
        {services.map((service, idx) => (
          <Card
            key={idx}
            className="hover:shadow-lg transition-transform hover:-translate-y-1"
          >
            <CardHeader className="p-4 md:p-6 flex flex-col justify-between h-full">
              <h3 className="text-lg mb-3 md:text-xl font-semibold text-[#2765c3]">
                {service.title}
              </h3>

              <div className="flex-grow" />

              <p className="text-sm mb-3 md:text-base">{service.description}</p>

              <p className="font-bold mt-2 md:mt-3 text-red-600">
                {service.price}
              </p>
            </CardHeader>
          </Card>
        ))}
      </div>
    </section>
  );
}
