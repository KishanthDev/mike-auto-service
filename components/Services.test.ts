import React from "react";
import { render, screen } from "@testing-library/react";
import ServicesSection from "./ServicesSection";
import { Service } from "@/types/data";

// Mock the Card components using React.createElement
jest.mock("@heroui/card", () => ({
  Card: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) =>
    React.createElement(
      "div",
      {
        className: `mock-card ${className}`,
        "data-testid": "service-card",
      },
      children,
    ),
  CardHeader: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) =>
    React.createElement(
      "div",
      {
        className: `mock-card-header ${className}`,
        "data-testid": "card-header",
      },
      children,
    ),
}));

describe("ServicesSection Component", () => {
  const mockServices: Service[] = [
    {
      title: "Oil Change",
      description: "Complete oil and filter replacement",
      price: "$49.99",
    },
    {
      title: "Brake Service",
      description: "Brake inspection and pad replacement",
      price: "$129.99",
    },
  ];

  beforeEach(() => {
    render(React.createElement(ServicesSection, { services: mockServices }));
  });


  it("displays the correct heading", () => {
    const heading = screen.getByRole("heading", { level: 2 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent("Our Services");
    expect(heading).toHaveClass("text-2xl");
    expect(heading).toHaveClass("border-b-2");
    expect(heading).toHaveClass("border-red-600");
    expect(heading).toHaveClass("text-[#2765c3]");
  });

  it("renders all service cards", () => {
    const cards = screen.getAllByTestId("service-card");
    expect(cards).toHaveLength(mockServices.length);
  });

  it("applies correct card styling and hover effects", () => {
    const cards = screen.getAllByTestId("service-card");
    cards.forEach((card) => {
      expect(card).toHaveClass("hover:shadow-lg");
      expect(card).toHaveClass("transition-transform");
      expect(card).toHaveClass("hover:-translate-y-1");
    });
  });

  it("displays correct service information", () => {
    mockServices.forEach((service) => {
      expect(screen.getByText(service.title)).toBeInTheDocument();
      expect(screen.getByText(service.description)).toBeInTheDocument();
      expect(screen.getByText(service.price)).toBeInTheDocument();
    });
  });

  it("applies correct responsive grid layout", () => {
    const grid = screen.getByTestId("services-grid");
    expect(grid).toHaveClass("grid");
    expect(grid).toHaveClass("grid-cols-1");
    expect(grid).toHaveClass("sm:grid-cols-2");
    expect(grid).toHaveClass("gap-4");
  });

  it("applies correct header styling", () => {
    const headers = screen.getAllByTestId("card-header");
    headers.forEach((header) => {
      expect(header).toHaveClass("p-4");
      expect(header).toHaveClass("md:p-6");
      expect(header).toHaveClass("flex");
      expect(header).toHaveClass("flex-col");
    });
  });

  it("applies correct title styling", () => {
    const titles = screen.getAllByRole("heading", { level: 3 });
    titles.forEach((title) => {
      expect(title).toHaveClass("text-lg");
      expect(title).toHaveClass("md:text-xl");
      expect(title).toHaveClass("text-[#2765c3]");
    });
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      React.createElement(ServicesSection, { services: mockServices }),
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
