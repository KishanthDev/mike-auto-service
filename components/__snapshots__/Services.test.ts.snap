// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesSection Component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#2765c3]"
    >
      Our Services
    </h2>
    <div
      class="grid grid-cols-1 sm:grid-cols-2 gap-4 mt-4 md:mt-6"
      data-testid="services-grid"
    >
      <div
        class="mock-card hover:shadow-lg transition-transform hover:-translate-y-1"
        data-testid="service-card"
      >
        <div
          class="mock-card-header p-4 md:p-6 flex flex-col justify-between h-full"
          data-testid="card-header"
        >
          <h3
            class="text-lg mb-3 md:text-xl font-semibold text-[#2765c3]"
          >
            Oil Change
          </h3>
          <div
            class="flex-grow"
          />
          <p
            class="text-sm mb-3 md:text-base"
          >
            Complete oil and filter replacement
          </p>
          <p
            class="font-bold mt-2 md:mt-3 text-red-600"
          >
            $49.99
          </p>
        </div>
      </div>
      <div
        class="mock-card hover:shadow-lg transition-transform hover:-translate-y-1"
        data-testid="service-card"
      >
        <div
          class="mock-card-header p-4 md:p-6 flex flex-col justify-between h-full"
          data-testid="card-header"
        >
          <h3
            class="text-lg mb-3 md:text-xl font-semibold text-[#2765c3]"
          >
            Brake Service
          </h3>
          <div
            class="flex-grow"
          />
          <p
            class="text-sm mb-3 md:text-base"
          >
            Brake inspection and pad replacement
          </p>
          <p
            class="font-bold mt-2 md:mt-3 text-red-600"
          >
            $129.99
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
